### ISSUE TITLE
Error: Undefined property $columns on hasManyThrough relationship

### ISSUE THREAD

**Issue by sameerpanjwani** [yajra/laravel-datatables-oracle#28]
There seems to be an issue when working with hasManyThrough relationships. This is my query:

```php
$campaigns = User::find($user_id)->campaignDetails()->select(['user_campaigns.user_id','campaign_details.campaign_id as as','campaign_name','campaign_status']);
        return Datatables::of($campaigns)->make(true);

public function campaignDetails()
    {
        return $this->hasManyThrough('App\CampaignDetail','App\UserCampaign','user_id','campaign_id');
    }
```

This is the code that seems to be causing the problem. Line 298 in Datatables\Datatables.php:
`$this->columns = $this->query_type == 'eloquent' ? $this->query->getQuery()->columns : $this->query->columns;`

But it seems that Eloquent is not returning you the columns and that's why I'm guessing it's failing? Any workaround?

**Comment by yajra** [comment]
Hi, haven't tried the package via `HasMany` and the likes but will surely check this out on my free time.

In the meantime, the workaround you can do is by using a `join` statement manually like below: Just adjust it according to your tables definition.

```php
$campaigns = Campaigns::join('user_campaigns','user_campaigns.campaign_id','=','campaigns.id')
	->join('campaign_details','campaign_details.id','=','user_campaigns.details_id')
	->select([
		'user_campaigns.user_id',
		'campaign_details.campaign_id as as',
		'campaign_name',
		'campaign_status'
	])
	->where('campaigns.user_id', $user_id);

return Datatables::of($campaigns)->make(true);
```

**Comment by sameerpanjwani** [comment]
Thanks, will try it. Also noticed that there are two aggregate count
queries running instead of one for all types of queries.

**Comment by yajra** [comment]
Yes, two count query is required. 1 for the total records and 1 for the total filtered records.

**Comment by yajra** [comment]
Should be fixed on [v4.3.0](e6cc9c541da310eff69e8b65ad85291907d39285). Kindly checkout and test. Thanks!

**Comment by sameerpanjwani** [comment]
Yes works, thank you!
